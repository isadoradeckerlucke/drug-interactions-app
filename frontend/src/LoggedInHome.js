import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { resetLoading } from "./actions/resetLoading";
import { toggleLoading } from "./actions/loading";
import { getInteractionsFromAPI } from "./actions/drugs";
import InteractionList from "./drugs/InteractionList";
import LoadingPage from "./LoadingPage";

function LoggedInHome(currentUser) {
  const interactions = useSelector((st) => st.drugs.interactions);
  const loading = useSelector((st) => st.loading.loading);
  const dispatch = useDispatch();

  const savedDrugsString = currentUser.currentUser.currentUser.savedDrugs.join(
    ", "
  );

  // get interactions with the users saved drugs, and toggle loading to true
  useEffect(() => {
    dispatch(getInteractionsFromAPI(savedDrugsString));
    dispatch(toggleLoading());
  }, [dispatch, savedDrugsString]);

  // when the interactions are loaded, reset loading to false
  useEffect(() => {
    if (interactions) {
      dispatch(resetLoading());
    }
  }, [dispatch, interactions]);

  return (
    <div>
      <h2>
        Welcome to the Drug Interaction Hub,{" "}
        {currentUser.currentUser.currentUser.username}
      </h2>
      <h4>interactions with your saved medications:</h4>
      {loading ? (
        <LoadingPage searchTerm={savedDrugsString}></LoadingPage>
      ) : (
        <p id="error-holder"></p>
      )}
      {interactions ? (
        Object.keys(interactions).length !== 0 ? (
          <InteractionList interactions={interactions} />
        ) : (
          <p>sorry, no results were found</p>
        )
      ) : (
        <p></p>
      )}{" "}
    </div>
  );
}

export default LoggedInHome;
