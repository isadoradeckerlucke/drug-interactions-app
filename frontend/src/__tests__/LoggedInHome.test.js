import React from "react";
import { render } from "@testing-library/react";
import LoggedInHome from "../LoggedInHome";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("home page works as expected", function () {
  const currentUserTest = {
    isLoggedIn: true,
    currentUser: {
      username: "testuser",
      email: "testuser@gmail.com",
      savedDrugs: [],
    },
  };
  const initialState = {
    searchTerm: { searchTerm: "adderall" },
    currentUser: currentUserTest,
    drugs: { interactions: {} },
    loading: { loading: false },
  };
  const middlewares = [thunk];
  const mockStore = configureStore(middlewares);
  let store;

  /** smoke test */
  it("renders without crashing, logged in user", function () {
    store = mockStore(initialState);
    render(
      <MemoryRouter>
        <Provider store={store}>
          <LoggedInHome currentUser={currentUserTest} />
        </Provider>
      </MemoryRouter>
    );
  });

  /** snapshot test */
  it("matches snapshot, logged in user", function () {
    store = mockStore(initialState);
    const { asFragment } = render(
      <MemoryRouter>
        <Provider store={store}>
          <LoggedInHome currentUser={currentUserTest} />
        </Provider>
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
