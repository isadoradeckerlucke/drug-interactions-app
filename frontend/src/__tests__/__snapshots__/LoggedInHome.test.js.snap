// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home page works as expected matches snapshot, logged in user 1`] = `
<DocumentFragment>
  <div>
    <h2>
      Welcome to the Drug Interaction Hub, testuser
    </h2>
    <h4>
      interactions with your saved medications:
    </h4>
    <p
      id="error-holder"
    />
    <p>
      sorry, no results were found
    </p>
     
  </div>
</DocumentFragment>
`;
